src/distributions/uniform.rs

```diff
@@ -259,20 +263,61 @@ pub trait UniformSampler: Sized {
         let uniform: Self = UniformSampler::new(low, high);
         uniform.sample(rng)
     }
+
+    /// Sample a single value uniformly from a range with inclusive lower bound
+    /// and inclusive upper bound `[low, high]`.
+    ///
+    /// By default this is implemented using
+    /// `UniformSampler::new_inclusive(low, high).sample(rng)`. However, for
+    /// some types more optimal implementations for single usage may be provided
+    /// via this method.
+    /// Results may not be identical.
+    fn sample_single_inclusive<R: Rng + ?Sized, B1, B2>(low: B1, high: B2, rng: &mut R)
+        -> Self::X
+        where B1: SampleBorrow<Self::X> + Sized,
+              B2: SampleBorrow<Self::X> + Sized
+    {
+        let uniform: Self = UniformSampler::new_inclusive(low, high);
+        uniform.sample(rng)
```

Comment:
It may make more sense to convert `sample_single` into `sample_single_inclusive`, then implement `sample_single` via `sample_single_inclusive`. At any rate, we *should* have an optimised implementation of this function.

Reply:
> implement sample_single via sample_single_inclusive

Is there a way to do this without introducing an awkward `where B2: From<u8> + Sub<B2>` constraint?